### DISCUSSION TITLE
A way to remove specific indexes from dataset

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#6883] [Q&A] [answered]
Is there a way to remove specific graphs from my dataset. What I am trying to do is predict on my test set, and then remove anything from the set that was predicted 0, then concatenate the predicted 1s with the training set creating a NEW training set. Thoughts?

**Comment by anthonysirico** [comment]
UPDATE: I am trying to use `.index_select` to create another set from a list of indices. If `ones = [1 5 6 9 7 10 15 16 ... 22546]` and I use the line 
```
ones_set = test_set.index_select(ones)
```
it returns an out of range error. I have tried converting the list to an ndarray as the docs said to use and a tensor of type long, and it keeps throwing the same error.

Again, the end goal is to take specific indices out of a set to create another set, to later concatenate it with the training set.

**Reply by wsad1** [reply]
Is `len(test_set)` larger than `max(ones)`? Also could you paste the full error message you are seeing?

**Comment by anthonysirico** [comment, marked as answer]
Figured it out: used a for loop that cycled through the indexes in the list.
